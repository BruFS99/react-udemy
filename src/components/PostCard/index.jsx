import P from 'prop-types';
import React  from 'react';
import '../PostCard/styles.css';

export const PostCard = ({id, cover, title, body}) => {

    return (
        <div className='post' key={id}>
            <img src={cover} alt={title} />
            <div className="post-content">
                <h1>{title} {id}</h1>
                <p>{body}</p>
            </div>
        </div>
    )

};

PostCard.propTypes = {
  id: P.number.isRequired,
  cover: P.string.isRequired,
  title: P.string.isRequired,
  body: P.string.isRequired,
};
