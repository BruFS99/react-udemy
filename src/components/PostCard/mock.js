export const postCardPropsMock = {
    id: 1,
    title: 'Titulo teste',
    cover: 'img.img.png',
    body: 'Body teste'
}