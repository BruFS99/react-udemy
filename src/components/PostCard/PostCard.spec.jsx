import { render, screen } from "@testing-library/react"
import { PostCard } from "../PostCard"
import { postCardPropsMock } from "./mock"

const props = postCardPropsMock

describe('<PostCard/>', () => {
    it('should render PostCard correctly', () => {
       render(<PostCard {...props}/>);

        expect(screen.getByAltText(/Titulo teste/i)).toHaveAttribute('src', 'img.img.png');
        expect(screen.getByRole('heading', {name: 'Titulo teste 1'})).toBeInTheDocument();
        expect(screen.getByText('Body teste')).toBeInTheDocument();

        expect.assertions(3)

    })

    it('should match snapshot', () => {
        const {container} = render(<PostCard {...props}/>);
 
         expect(container.firstChild).toMatchSnapshot();
     })
})