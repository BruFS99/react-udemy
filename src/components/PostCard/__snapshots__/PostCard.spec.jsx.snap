// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard/> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Titulo teste"
    src="img.img.png"
  />
  <div
    class="post-content"
  >
    <h1>
      Titulo teste
       
      1
    </h1>
    <p>
      Body teste
    </p>
  </div>
</div>
`;
